src/auto-evo/RunResults.cs

```diff
@@ -58,9 +58,12 @@ public void ApplyResults(GameWorld world, bool skipMutations)
 
                     if (patch != null)
                     {
-                        // We ignore the return value as population results are added for all existing patches for all
-                        // species (if the species is not in the patch the population is 0 in the results)
-                        patch.UpdateSpeciesPopulation(entry.Key, populationEntry.Value);
+                        // Check if species is extinct and remove it from the game.
+                        if (patch.UpdateSpeciesPopulation(entry.Key, populationEntry.Value)
+                            && populationEntry.Value <= 0)
+                        {
+                            world.RemoveSpecies(entry.Key);
```

Comment:
I think that this doesn't work correctly. A species should only be made extinct when it is gone from *all* the patches.

Reply:
Sorry, I have not thought of that. The code now checks every patch first.